doc/dev/tests.md

```diff
@@ -0,0 +1,295 @@
+- [Writing New Tests](#Writing-new-tests)
+- [Define our credentials and settings](#define-credentials)
+- [Create live test resources](#create-live-test-resources)
+- [Write our test](#writing-your-test)
+- [An example test](#an-example-test)
+- [Run and record our tests](#run-and-record-the-test)
+
+## Setup your development environment
+
+The Azure SDK Python team creates libraries that are compatible with Python 2.7 and 3.5 and up. We will set up working Python virtual environments for Python 2.7, 3.5, and 3.9. It is recommended to do your development work in Python3, however it is helpful to have virtual environments for other versions to make debugging PRs easier locally.
+
+* Python 3.9: Use the [python website](https://www.python.org/downloads/) or the one-click experience from the [Windows store](https://www.microsoft.com/p/python-39/9p7qfqmjrfp7) (Windows only).
+* Python 3.5: Use the [python website](https://www.python.org/downloads/release/python-3510/)
+* Python 2.7: Use the [python website](https://www.python.org/downloads/release/python-2718/)
+```cmd
+C:\Users> python -m venv env
+C:\Users> env\scripts\activate       # PowerShell only
+C:\Users> source env\bin\activate    # Linux shell (Bash, ZSH, etc.) only
+C:\Users> env\scripts\activate.bat   # Windows CMD only
+(env)C:\Users>
+```
+To create virtual environment for different versions of Python use the `-p` flag to pass the specific Python executable you want to use
+```cmd
+C:\Users> python -m venv -p <path/to/Python/Python35/python.exe> py35_venv
+C:\Users> python -m venv -p <path/to/Python/Python27/python.exe> py27_venv
+```
+
+### SDK root directory
+
+In the root directory of our SDK, a number of mandatory files have been added. You will not need to edit these files for this tutorial, but you should be aware about the various components and where to find more details on them. When creating your own SDK, these files can be copied from the [`sdk/template`](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/template) project, and modified to your needs.
+
+- README.md. This is the description and guidance for customers or your SDK. Please see the guide on writing a README to make sure you have the complete [content requirements and formatting](https://review.docs.microsoft.com/help/contribute-ref/contribute-ref-how-to-document-sdk?branch=master#readme).
+- CHANGELOG.md. This is where you will add the summary of changes for each new release. Please see [the guidance](https://azure.github.io/azure-sdk/policies_releases.html#changelog-guidance) for correct formatting.
+- setup.py. This is the 'installer' for your Python SDK. Please see [the guide on Python packaging](https://github.com/Azure/azure-sdk-pr/issues/225) for details on customizing this for a specific package.
+- setup.cfg. This is an artifact used in building the Python package. Please see [the guide on Python packaging](https://github.com/Azure/azure-sdk-pr/issues/225) for details.
```

Comment:
I understand this is work in progress, but then let's do a public open issue "todo" in our repo.,